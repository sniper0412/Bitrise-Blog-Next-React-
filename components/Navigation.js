import React from 'react';
import SVG from 'react-inlinesvg';
import cx from 'classnames';

const UTM = '? utm_source=bitriseblog&utm_medium=web&utm_campaign=static';

export default class Menu extends React.Component {
  state = {
    isMenuOpen: false
  };

  toggleOpen() {
    this.setState({ isMenuOpen: !this.state.isMenuOpen });
  }

  render() {
    const { isMenuOpen } = this.state;

    return (
      // <% environment.context_class.instance_eval { include ApplicationHelper } %>
      <nav className="menu">
        <div className="menu-content">
          <a className="logo" href="/">
            <SVG src="/static/svg/logo.svg" />
          </a>
          <ul className={cx('menu-options', { opened: isMenuOpen })}>
            <li className="menu-option">
              <a href={`https://www.bitrise.io/integrations${UTM}`}>Integrations</a>
            </li>
            <li className="menu-option">
              <a href={`https://www.bitrise.io/pricing${UTM}`}>Pricing</a>
            </li>
            <li className="menu-option">
              <a href={`https://www.bitrise.io/cli${UTM}`}>CLI</a>
            </li>
            <li className="menu-option">
              <a href={`https://www.bitrise.io/contact${UTM}`}>Contact</a>
            </li>
            <li className="menu-option">
              <a href={`http://devcenter.bitrise.io${UTM}`}>Docs</a>
            </li>
            <li className="side-menu-options">
              <ul>
                <li className="side-menu-option login">
                  <a href={`https://www.bitrise.io/users/sign_in${UTM}`}>Log in</a>
                </li>
                <li className="side-menu-option signup">
                  <a href={`https://www.bitrise.io/users/sign_in${UTM}`}>Sign up!</a>
                </li>
              </ul>
            </li>
          </ul>
          <button className={cx('burger', { open: isMenuOpen })} onClick={() => this.toggleOpen()}>
            <SVG src="/static/svg/burger.svg" />
          </button>
        </div>
      </nav>
    );
  }
}
