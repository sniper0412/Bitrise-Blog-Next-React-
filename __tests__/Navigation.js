import renderer from 'react-test-renderer';

import Navigation from '../components/Navigation';

describe('Navigation component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Navigation />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with burger menu clicked', () => {
    const nav = renderer.create(<Navigation />);
    nav.root.findByType('button').props.onClick();
    expect(nav.toJSON()).toMatchSnapshot();
  });
});
