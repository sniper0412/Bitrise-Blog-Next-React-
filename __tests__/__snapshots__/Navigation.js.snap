// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<nav
  className="menu"
>
  <div
    className="menu-content"
  >
    <a
      className="logo"
      href="/"
    >
      <span
        className="isvg loading"
      />
    </a>
    <ul
      className="menu-options"
    >
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/integrations? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Integrations
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/pricing? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Pricing
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/cli? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          CLI
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/contact? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Contact
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="http://devcenter.bitrise.io? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Docs
        </a>
      </li>
      <li
        className="side-menu-options"
      >
        <ul>
          <li
            className="side-menu-option login"
          >
            <a
              href="https://www.bitrise.io/users/sign_in? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
            >
              Log in
            </a>
          </li>
          <li
            className="side-menu-option signup"
          >
            <a
              href="https://www.bitrise.io/users/sign_in? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
            >
              Sign up!
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <button
      className="burger"
      onClick={[Function]}
    >
      <span
        className="isvg loading"
      />
    </button>
  </div>
</nav>
`;

exports[`Navigation component renders correctly with burger menu clicked 1`] = `
<nav
  className="menu"
>
  <div
    className="menu-content"
  >
    <a
      className="logo"
      href="/"
    >
      <span
        className="isvg loading"
      />
    </a>
    <ul
      className="menu-options opened"
    >
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/integrations? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Integrations
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/pricing? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Pricing
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/cli? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          CLI
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="https://www.bitrise.io/contact? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Contact
        </a>
      </li>
      <li
        className="menu-option"
      >
        <a
          href="http://devcenter.bitrise.io? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
        >
          Docs
        </a>
      </li>
      <li
        className="side-menu-options"
      >
        <ul>
          <li
            className="side-menu-option login"
          >
            <a
              href="https://www.bitrise.io/users/sign_in? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
            >
              Log in
            </a>
          </li>
          <li
            className="side-menu-option signup"
          >
            <a
              href="https://www.bitrise.io/users/sign_in? utm_source=bitriseblog&utm_medium=web&utm_campaign=static"
            >
              Sign up!
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <button
      className="burger open"
      onClick={[Function]}
    >
      <span
        className="isvg loading"
      />
    </button>
  </div>
</nav>
`;
